htdocs/bom/class/api_boms.class.php

```diff
@@ -279,6 +280,36 @@ public function delete($id)
 		);
 	}
 
+	/**
+	 * Get lines of an BOM
+	 *
+	 * @param int   $id             Id of BOM
+	 *
+	 * @url	GET {id}/lines
+	 *
+	 * @return array
+	 */
+	public function getLines($id)
+	{
+		if (!DolibarrApiAccess::$user->rights->bom->read) {
+			throw new RestException(401);
+		}
+
+		$result = $this->bom->fetch($id);
+		if (!$result) {
+			throw new RestException(404, 'BOM not found');
+		}
+
+		if (!DolibarrApi::_checkAccessToResource('bom_bom', $this->commande->id)) {
```

Comment:
Something seems wrong here: Do you mean 
$this->bom->id  and not $this->commande->id